### PR TITLE
Classes

### PR DESCRIPTION
**Pull request by HabibPayenda**
### **The following requirements were added to the project:** 

- Create a class Person with the following:
- [ ] Instance vars: @id, @name, and @age.
- [ ] Constructor with name, age, and parent_permission as parameters. name and parent_permission are optional and have default values of "Unknown" and true.
- [ ] Getters for @id, @name, and @age.
- [ ] Setters for @name and @age.
- [ ] Private method is_of_age? that returns true if @age is greater or equal to 18 and false otherwise.
- [ ] Public method can_use_services? that returns true if a person is of age or if they have permission from parents.

- Create class Student with the following:
- [ ] Inherits from Person.
- [ ] Constructor extends parent's constructor by adding @classroom and a parameter for it.
- [ ] Method play_hooky that returns "¯\(ツ)/¯".

- Create a class Teacher with the following:
- [ ] Inherits from Person.
- [ ] The constructor extends the parent's constructor by adding @specialization and a parameter for it.
- [ ] Override can_use_services? so it always returns true.

- Each class should be saved in a separate file.

Thank you.

### CODE REVIEW

**Review by zieeco — CHANGES_REQUESTED** [review]
Hi @HabibPayenda,


Great job so far! 👏 
You've done well, however, there are some issues that you still need to work on to go to the next project, but you are almost there!


## To highlight 🎯

- No linter errors.✔️
- Descriptive PR. ✔️


## Required Changes ♻️
_Check the comments below._

- [ ] According to the [Project Milestone](https://github.com/microverseinc/curriculum-ruby/blob/main/oop/school_library_people.md), the `Student`, `Person`, and `Teacher` classes are empty and missing all the required attributes, methods, etc that is required for this project. please refer to the project requirement and ensure you've implemented them before submitting for another review.





## Optional suggestions
- N/A

**_Every comment with the **[OPTIONAL]** prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._**


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: person.rb [hunk 1]
```diff
@@ -1 +1,42 @@
...
+    @id
+  end
+
+  def age_writer(age)
+    @age = age
+  end
+
+  def name_writer(name)
+    @name = name
+  end
```
Comment (sja-thedude) [inline person.rb]: - [ ] All the highlighted code is not required instead you should add a getter & setter before initialize method.

**Review by sja-thedude — CHANGES_REQUESTED** [review]
## Required Changes ♻️

Hi @HabibPayenda, This is @sja-thedude your code reviewer.

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.

200w

 ## Highlights
✔️ Good job so far

 ## Required Changes ♻️
- [ ] _Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me [@sja-thedude] in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: person.rb (hunk 1 above)
Comment (HabibPayenda) [reply person.rb]: @sja-thedude Can you please provide some example code?

Comment (sja-thedude) [reply person.rb]: - [ ] Its just that you have to add getter & setter methods like this before initialize method.
```
attr_accessor :name, :age 
  attr_reader :id 
```

Comment (HabibPayenda) [reply person.rb]: @sja-thedude Thank you so much.

**Review by TedLivist — APPROVED** [review]
## STATUS: APPROVED ✔️ ✔️

Hi @HabibPayenda,

✔️ Good documentation

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. And please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**